hie-plugin-api/Haskell/Ide/Engine/TypeMap.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE LambdaCase #-}
+module Haskell.Ide.Engine.TypeMap where
+
+import qualified Data.IntervalMap.FingerTree   as IM
+
+import qualified GHC
+import           GHC                            ( TypecheckedModule )
+import           GhcMod.SrcUtils
+
+import           Data.Data                     as Data
+import           Control.Monad.IO.Class
+import           Data.Maybe
+import qualified TcHsSyn
+import qualified TysWiredIn
+import qualified CoreUtils
+import qualified Type
+import qualified Desugar
+
+import           Haskell.Ide.Engine.ArtifactMap
+
+-- | Generate a mapping from an Interval to types.
+-- Intervals may overlap and return more specific results.
+genTypeMap :: GHC.GhcMonad m => TypecheckedModule -> m TypeMap
+genTypeMap tm = do
+  let typecheckedSource = GHC.tm_typechecked_source tm
+  hs_env <- GHC.getSession
+  liftIO $ types hs_env typecheckedSource
+
+collectAllSpansTypes'
+  :: GHC.GhcMonad m => Bool -> TypecheckedModule -> m [(GHC.SrcSpan, GHC.Type)]
+collectAllSpansTypes' = collectAllSpansTypes
```

Comment:
What is the point of this function?

Reply:
This is a leftover, thanks for catching it, but it is probably in your codebase now as well.